pkg/statistics/handle/bootstrap.go

```diff
@@ -549,7 +585,9 @@ func (*Handle) initStatsBuckets4Chunk(cache statstypes.StatsCache, iter *chunk.I
 			}
 		}
 		hist.AppendBucketWithNDV(&lower, &upper, row.GetInt64(3), row.GetInt64(4), row.GetInt64(7))
-		cache.Put(tableID, table) // put this table again since it is updated
+	}
+	if table != nil {
```

Comment:
when we arrive here, it could be the end of single chunk iter, rather than the end of all result chunks?
if yes, the possibility that same physical table id records could be exists in next chunk iter

Reply:
It will check the table ID. if the table is not the same as the last one, it will be written into the stats cache.  this code is above here.

Reply:
okk